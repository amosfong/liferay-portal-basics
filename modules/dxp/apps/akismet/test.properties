##
## Test Suites
##

    modified.files.includes[relevant][akismet-functional-rule]=**

    modified.files.includes[relevant][akismet-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=\
        akismet-functional-rule,\
        akismet-java-rule

    test.batch.names[relevant][akismet-functional-rule]=\
        functional-tomcat90-mysql57

    test.batch.names[relevant][akismet-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]}

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][akismet-java-rule]=\
        dxp/apps/akismet/**/*Test.java

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][akismet-functional-rule]=\
        (testray.main.component.name ~ "Akismet")

##
## Testray
##

    testray.main.component.name=Akismet