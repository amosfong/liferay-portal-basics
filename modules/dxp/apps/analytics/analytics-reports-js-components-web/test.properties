##
## Test Suites
##

    modified.files.includes[relevant][analytics-reports-js-components-web-playwright-rule]=**
    relevant.rule.names=analytics-reports-js-components-web-playwright-rule

    test.batch.names[relevant][analytics-reports-js-components-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][analytics-reports-js-components-web-playwright-rule]=\
        analytics-reports-js-components-web