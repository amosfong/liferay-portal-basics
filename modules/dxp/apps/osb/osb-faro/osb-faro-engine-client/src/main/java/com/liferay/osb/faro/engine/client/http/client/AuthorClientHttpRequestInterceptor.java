/**
 * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.engine.client.http.client;

import com.liferay.osb.faro.engine.client.constants.OSBAsahHeaderConstants;
import com.liferay.portal.kernel.model.User;
import com.liferay.portal.kernel.security.permission.PermissionChecker;
import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;

import java.io.IOException;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpRequest;
import org.springframework.http.client.ClientHttpRequestExecution;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.http.client.ClientHttpResponse;

/**
 * @author Marcellus Tavares
 */
public class AuthorClientHttpRequestInterceptor
	implements ClientHttpRequestInterceptor {

	@Override
	public ClientHttpResponse intercept(
			HttpRequest httpRequest, byte[] bytes,
			ClientHttpRequestExecution clientHttpRequestExecution)
		throws IOException {

		_addAuthorHeaders(httpRequest);

		return clientHttpRequestExecution.execute(httpRequest, bytes);
	}

	private void _addAuthorHeaders(HttpRequest httpRequest) {
		User user = _getUser();

		if (user == null) {
			return;
		}

		HttpHeaders httpHeaders = httpRequest.getHeaders();

		httpHeaders.add(
			OSBAsahHeaderConstants.AUTHOR_USER_ID,
			String.valueOf(user.getUserId()));
		httpHeaders.add(
			OSBAsahHeaderConstants.AUTHOR_USER_NAME, user.getFullName());
	}

	private User _getUser() {
		PermissionChecker permissionChecker =
			PermissionThreadLocal.getPermissionChecker();

		if (permissionChecker == null) {
			return null;
		}

		return permissionChecker.getUser();
	}

}