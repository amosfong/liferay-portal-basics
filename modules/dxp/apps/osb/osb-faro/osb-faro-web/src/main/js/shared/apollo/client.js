import cache from './cache';
import {ApolloClient} from 'apollo-client';
import {DEVELOPER_MODE} from 'shared/util/constants';
import {get} from 'lodash';
import {HttpLink} from 'apollo-link-http';
import {onError} from 'apollo-link-error';
import {reloadPage} from 'shared/util/router';
import {resolvers} from './resolvers/resolvers';

const groupIdRegex = /^\/workspace\/([a-z0-9._-]+)/;

/**
 * Fecth With Project ID
 * @param {string} uri
 * @param {object} options
 */
const fetchWithGroupId = (uri, options) => {
	const currentUri = new URL(window.location.href);
	const {operationName} = JSON.parse(options.body);
	const matches = currentUri.pathname.match(groupIdRegex);

	if (matches !== null && matches.length > 1) {
		const groupId = matches[1];

		return fetch(
			`${uri}?opname=${operationName}&projectGroupId=${groupId}`,
			options
		);
	}

	return fetch(`${uri}/&opname=${operationName}`, options);
};

const client = new ApolloClient({
	addTypename: true,
	cache,
	connectToDevTools: FARO_DEV_MODE,
	defaultOptions: {
		watchQuery: {
			notifyOnNetworkStatusChange: true
		}
	},
	link: HttpLink.from([
		onError(({operation}) => {
			const status = get(operation.getContext(), ['response', 'status']);

			if (status === 401) {
				reloadPage();
			}
		}),
		new HttpLink({
			credentials: 'same-origin',
			fetch: fetchWithGroupId,
			uri: '/o/cerebro/graphql'
		})
	]),

	resolvers: {
		/**
		 * Queries must render only in the dev mode to avoid
		 * add unnecessary code in the final bundle
		 */
		Query: DEVELOPER_MODE ? resolvers : {}
	}
});

export default client;
