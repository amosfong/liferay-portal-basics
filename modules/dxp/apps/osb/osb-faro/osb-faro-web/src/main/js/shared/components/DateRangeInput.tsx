import ClayButton from '@clayui/button';
import ClayDropDown, {Align} from '@clayui/drop-down';
import ClayIcon from '@clayui/icon';
import DatePicker from './date-picker';
import getCN from 'classnames';
import Input from './Input';
import moment from 'moment';
import React, {useState} from 'react';
import {DatePickerRetentionPeriodHeader} from './DatePickerRetentionPeriodHeader';
import {DEFAULT_DATE_FORMAT} from 'shared/util/date';
import {formatDateWithTimezone} from './dropdown-range-key/utils';
import {isNil, noop} from 'lodash';
import {sub} from 'shared/util/lang';
import {useRetentionPeriod} from 'shared/hooks/useRetentionPeriod';
import {useTimeZone} from 'shared/hooks/useTimeZone';

const convertToMoment = (value: string, format): moment.Moment => {
	const date = moment(value, format);

	return date.isValid() ? date : null;
};

export type DateRange = {
	end: string;
	start: string;
};

export type MomentDateRange = {
	end: moment.Moment;
	start: moment.Moment;
};

interface IDateInputProps {
	className?: string;
	displayFormat?: string;
	format?: string;
	groupId?: string;
	id?: string;
	name?: string;
	onBlur?: (event?: FocusEvent) => void;
	onChange: (range: DateRange) => void;
	overlayAlignment?: string;
	showRetentionPeriod?: boolean;
	usePortal?: boolean;
	value: DateRange;
}

const DateInput: React.FC<IDateInputProps> = ({
	className,
	displayFormat,
	format = DEFAULT_DATE_FORMAT,
	groupId,
	onBlur = noop,
	onChange = noop,
	showRetentionPeriod = true,
	value
}) => {
	const [active, setActive] = useState(false);

	const {timeZoneId} = useTimeZone(groupId);
	const retentionPeriod = useRetentionPeriod();

	const convertMomentToDisplayFormat = (value: moment.Moment): string =>
		isNil(value) ? null : value.format(displayFormat || format);

	const handleDateSelect = ({end, start}: MomentDateRange) => {
		onChange({
			end: convertMomentToDisplayFormat(end),
			start: convertMomentToDisplayFormat(start)
		});
	};

	const getDateRangeDisplay = ({end, start}: MomentDateRange): string => {
		if (end || start) {
			return sub(Liferay.Language.get('x-to-x'), [
				convertMomentToDisplayFormat(start),
				convertMomentToDisplayFormat(end)
			]) as string;
		}

		return '';
	};

	const momentDateRange = {
		end: convertToMoment(value.end, format),
		start: convertToMoment(value.start, format)
	};

	const minDate = formatDateWithTimezone(timeZoneId).clone();

	return (
		<ClayDropDown
			active={active}
			alignmentPosition={Align.TopLeft}
			className={getCN(className, 'dropdown-range-key-root')}
			menuElementAttrs={{
				className: getCN('dropdown-range-key-menu-root', {
					'show-date-picker': active
				}),
				style: {
					zIndex: 1060
				}
			}}
			onActiveChange={active => {
				setActive(active);

				!active && onBlur();
			}}
			trigger={
				<div>
					<Input.Group>
						<Input.GroupItem>
							<Input
								autoComplete='off'
								data-testid='date-range-input'
								inset='after'
								onClick={() => setActive(true)}
								placeholder={sub(
									Liferay.Language.get('x-to-x'),
									[
										Liferay.Language.get('yyyy-mm-dd'),
										Liferay.Language.get('yyyy-mm-dd')
									]
								)}
								readOnly
								value={getDateRangeDisplay(momentDateRange)}
							/>

							<Input.Inset position='after'>
								<ClayButton
									aria-label={Liferay.Language.get(
										'choose-date-range'
									)}
									className='button-root'
									displayType='unstyled'
									onClick={() => setActive(true)}
								>
									<ClayIcon
										className='icon-root'
										symbol='calendar'
									/>
								</ClayButton>
							</Input.Inset>
						</Input.GroupItem>
					</Input.Group>
				</div>
			}
		>
			<DatePicker
				date={momentDateRange}
				header={
					showRetentionPeriod ? (
						<DatePickerRetentionPeriodHeader
							retentionPeriod={retentionPeriod}
						/>
					) : null
				}
				maxDate={formatDateWithTimezone(timeZoneId)
					.clone()
					.subtract(1, 'days')}
				maxRange={365}
				minDate={
					showRetentionPeriod
						? minDate.subtract(retentionPeriod, 'months')
						: minDate.subtract(100, 'years')
				}
				onSelect={handleDateSelect}
				timeZoneId={timeZoneId}
			/>
		</ClayDropDown>
	);
};

export default DateInput;
