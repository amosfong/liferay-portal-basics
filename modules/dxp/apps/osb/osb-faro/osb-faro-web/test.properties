##
## Test Suites
##

    modified.files.includes[relevant][osb-faro-web-playwright-rule]=**
    relevant.rule.names=osb-faro-web-playwright-rule

    test.batch.names[relevant][osb-faro-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][osb-faro-web-playwright-rule]=\
        osb-faro-web

##
## Testray
##

    testray.main.component.name=Analytics Cloud