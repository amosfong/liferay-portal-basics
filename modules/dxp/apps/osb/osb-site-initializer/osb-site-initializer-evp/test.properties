##
## Test Suites
##

    modified.files.includes[relevant][osb-site-initializer-evp-functional-rule]=**
    relevant.rule.names=osb-site-initializer-evp-functional-rule

    test.batch.names[relevant][osb-site-initializer-evp-functional-rule]=\
        functional-tomcat90-mysql57

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP