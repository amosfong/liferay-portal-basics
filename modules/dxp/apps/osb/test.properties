##
## Test Suites
##

    modified.files.includes[relevant][osb-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=osb-java-rule

    test.batch.names[relevant][osb-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

##
## Testray
##

    testray.main.component.name=Analytics Cloud