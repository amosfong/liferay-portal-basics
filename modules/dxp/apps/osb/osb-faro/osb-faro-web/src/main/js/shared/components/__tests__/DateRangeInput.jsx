import client from 'shared/apollo/client';
import DateRangeInput from '../DateRangeInput';
import mockStore from 'test/mock-store';
import moment from 'moment';
import React from 'react';
import {ApolloProvider} from '@apollo/react-hooks';
import {cleanup, render} from '@testing-library/react';
import {MockedProvider} from '@apollo/react-testing';
import {mockPreferenceReq} from 'test/graphql-data';
import {Provider} from 'react-redux';

jest.unmock('react-dom');

jest.mock('shared/hooks/useTimeZone', () => ({
	useTimeZone: () => ({timeZoneId: 'UTC'})
}));

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useParams: () => ({
		channelId: '456',
		groupId: '2000',
		query: {
			rangeKey: '30'
		}
	})
}));

describe('DateRangeInput', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(
			<ApolloProvider client={client}>
				<Provider store={mockStore()}>
					<MockedProvider mocks={[mockPreferenceReq()]}>
						<DateRangeInput
							value={{
								end: moment(100000000000),
								start: moment(0)
							}}
						/>
					</MockedProvider>
				</Provider>
			</ApolloProvider>
		);

		expect(container).toMatchSnapshot();
	});
});
