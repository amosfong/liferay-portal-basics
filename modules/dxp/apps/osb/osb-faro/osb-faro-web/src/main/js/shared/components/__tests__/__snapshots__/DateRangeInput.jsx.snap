// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeInput renders 1`] = `
<div>
  <div
    class="dropdown dropdown-range-key-root"
  >
    <div
      aria-controls="clay-dropdown-menu-1"
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle"
    >
      <div
        class="input-group"
      >
        <div
          class="input-group-item"
        >
          <input
            autocomplete="off"
            class="input-root form-control input-group-inset input-group-inset-after"
            data-testid="date-range-input"
            placeholder="YYYY-MM-DD to YYYY-MM-DD"
            readonly=""
            type="text"
            value="1970-01-01 to 1973-03-03"
          />
          <div
            class="input-group-inset-item input-group-inset-item-after"
          >
            <button
              aria-label="Choose Date Range"
              class="button-root btn btn-unstyled"
              type="button"
            >
              <svg
                class="lexicon-icon lexicon-icon-calendar icon-root"
                role="presentation"
              >
                <use
                  href="#calendar"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
